### ISSUE TITLE
If a collection has never had a model saved successfully to the server, it won't cache failed saves to localStorage

### ISSUE THREAD

**Issue by carljm** [nilbus/Backbone.dualStorage#130]
It seems that the key in `localStorage` for a collection is only created on a successful save to the server. So if a new collection is created with a never-before-used `storeName`, and then we go offline before ever saving a model in that collection, and then we save some models in the collection, those models are not synced to `localStorage` and no `_dirty` key is ever created, so the saves are basically lost. This is because `options.storeExists` is set to `false` at the top of `dualsync()`.

I'm not sure what the right fix is here. I'm not sure what the purpose of `options.storeExists` is in the first place, or why I'd ever want `dualStorage` to discard saves rather than creating a previously-nonexistent `localStorage` key for the collection.

I am currently working around this issue successfully by just having my collection do this in `initialize`:

```
          if (localStorage.getItem(this.storeName) === null) {
            localStorage.setItem(this.storeName, '');
          }
```

**Comment by carljm** [comment]
Any feedback on this issue? I'm happy to create a pull request to fix it, but since it's my first bug on the project, I'd be interested in confirmation that it's an actual bug, and some guidance on what sort of fix would be mergeable. Thanks!

**Comment by nilbus** [comment]
`storeExists` came into being in #94 for this scenario: Initially when there is no store, on fetch, it makes more sense to call the error callback than the success callback with an empty collection. This should [only affect the read method](https://github.com/nilbus/Backbone.dualStorage/blob/3505ba0/backbone.dualstorage.coffee#L283) (fetch). In addition to a successful save, the store is also created on a successful read [[1](https://github.com/nilbus/Backbone.dualStorage/blob/3505ba0/backbone.dualstorage.coffee#L323), [2](https://github.com/nilbus/Backbone.dualStorage/blob/3505ba0/backbone.dualstorage.coffee#L197)].

I actually have [tests in the mocha branch](https://github.com/nilbus/Backbone.dualStorage/blob/mocha/spec/basic_persistence_spec.coffee#L251) that cover this initial offline scenario with save (create and update) and fetch. 

That's my understanding from the point of view of the code. If there's a subtle bug I'm missing, then I'm glad you've found it. Given your workaround that fixes the problem, it seems like you're right. I welcome a pull request if you're able to find the source of the problem.

As for pull request guidelines—just have the latest version of coffee-script installed and run `make` before you make each commit, so all the relevant files get compiled.

**Comment by carljm** [comment]
Oops, this is my fault for not realizing that we're using older code, and not checking the latest code to see if it's already fixed. You fixed my issue quite a while ago, in https://github.com/nilbus/Backbone.dualStorage/commit/fde70e1422f28f368a957b23edb315607fc7488d

Sorry for the noise!

**Comment by nilbus** [comment]
Great!

**Comment by carljm** [comment]
I think actually I did check if there was a more recent release, but it seems the last release (1.3.1) was last May, before that fix. Any chance there will be a new release issued soon?

**Comment by nilbus** [comment]
Yes​, it's long overdue. Coming right up!

**Comment by nilbus** [comment]
Released in v1.4.0.

**Comment by carljm** [comment]
Awesome, thanks!
